examples/tree_1d_dgsem/elixir_shallowwater_multilayer_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,199 @@
+
+###############################################################################
+# Semidiscretization of the two-layer shallow water equations to test well-balancedness
+
+equations = ShallowWaterMultiLayerEquations1D(gravity_constant = 1.0, H0 = 2.0,
+                                              rhos = (1.0, 3.0))
+
+"""
+    initial_condition_twolayer_well_balanced_wet_dry(perturbation, equations:: ShallowWaterMultiLayerEquations1D)
+    initial_condition_twolayer_well_balanced_wet_dry(x, t, equations:: ShallowWaterMultiLayerEquations1D)
+
+Initial condition with a complex (discontinuous) bottom topography to test well-balancedness for a
+two-layer shallow water system with dry states if `perturbation` is set to `false`.
+Additionally, it is possible to set a perturbation in the lower layer to test perturbations from the 
+lake-at-rest condition by setting the `perturbation` variable to `true`.
+
+The initial condition is taken from the paper:
+  - S. Martínez-Aranda, A. Ramos-Pérez, P. García-Navarro (2020)
+    A 1D shallow-ﬂow model for two-layer ﬂows based on FORCE scheme with wet–dry treatment\n 
+    [DOI:10.2166/hydro.2020.002](https://doi.org/10.2166/hydro.2020.002)
+"""
+function initial_condition_twolayer_well_balanced_wet_dry(perturbation::Bool,
+                                                          equations::ShallowWaterMultiLayerEquations1D)
+    return function initial_condition_twolayer_well_balanced_wet_dry(x, t, equations)
+        # Set interface height for the upper layer
+        H_upper = 2.0
+
+        # Set interface height for the lower layer
+        H_lower = 1.0
+        if perturbation == true
+            if x[1] <= 5.0
+                nothing
+            elseif x[1] <= 15.0
+                H_lower = 0.75
+            elseif x[1] <= 35.0
+                H_lower = 1.5 - (x[1] - 15.0) / 20.0 * 0.5
+            elseif x[1] <= 40.0
+                H_lower = 1.0 + (x[1] - 35.0) / 5.0 * 0.5
+            elseif x[1] <= 55.0
+                H_lower = 1.0
+            elseif x[1] <= 70.0
+                H_lower = 1.6 - (x[1] - 55.0) / 15.0 * 0.4
+            elseif x[1] <= 75.0
+                nothing
+            elseif x[1] <= 85.0
+                H_lower = 1.0 - (x[1] - 75.0) / 10.0 * 0.3
+            elseif x[1] <= 95.0
+                H_lower = 1.1
+            else
+                H_lower = 1.5
+            end
+        end
+
+        # Set the bottom topography
+        if x[1] <= 5.0
+            b = 2.5
+        elseif x[1] <= 20.0
+            b = 0.5
+        elseif x[1] <= 40.0
+            b = 0.1 + (x[1] - 20.0) / 20.0 * 1.4
+        elseif x[1] <= 60.0
+            b = 1.0 - (x[1] - 40.0) / 20.0 * 1.3
+        elseif x[1] <= 70.0
+            b = 0.7
+        elseif x[1] <= 75.0
+            b = 2.2
+        elseif x[1] <= 95.0
+            b = 0.4
+        else
+            b = 1.5
+        end
+
+        # Set zero initial velocity
+        v1_upper = 0.0
+        v1_lower = 0.0
```

Comment:
```suggestion
        v1_upper = zero(H_upper)
        v1_lower = zero(H_upper)
```

Reply:
Done in 3d6d89a9381cd544f4ee470550c38ee47df42bf1